doc/releases/changelog-dev.md

```diff
@@ -129,6 +129,9 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* `dynamic_one_shot` was broken for old-API devices since `override_shots` was deprecated.
+  [(#6024)](https://github.com/PennyLaneAI/pennylane/pull/6024)
+
```

Comment:
Very minor thought - this might not count as a bug fix since the deprecation was started in this release cycle :thinking:

Reply:
Would you move it in deprecations?

Reply:
I'll merge to unblock Lightning PRs and fix in a follow-up if needed. Let's keep the discussion going nevertheless.